stl/inc/xspan

```diff
@@ -0,0 +1,238 @@
+    using element_type    = _Ty;
+    using value_type      = remove_cv_t<_Ty>;
+    using index_type      = size_t;
+    using difference_type = ptrdiff_t;
+    using pointer         = _Ty*;
+    using const_pointer   = const _Ty*;
+    using reference       = _Ty&;
+    using const_reference = const _Ty&;
+    // using iterator               = _Span_iterator<pointer>;
+    // using const_iterator         = _Span_const_iterator<const_pointer>;
+    // using reverse_iterator       = _STD reverse_iterator<iterator>;
+    // using const_reverse_iterator = _STD reverse_iterator<const_iterator>;
+
+    // [span.cons], span constructors, copy, assignment, and destructor
+    constexpr span() noexcept : _Mypair(_Zero_then_variadic_args_t(), nullptr) {}
+
+    constexpr span(pointer _Ptr, index_type _Count) noexcept : _Mypair(_One_then_variadic_args_t(), _Count, _Ptr) {}
+
+    constexpr span(pointer _First, pointer _Last) noexcept
+        : _Mypair(_One_then_variadic_args_t(), static_cast<index_type>(distance(_First, _Last)), _First) {}
+
+    template <size_t _Size>
+    constexpr span(element_type (&_Data)[_Size]) noexcept
+        : _Mypair(_One_then_variadic_args_t(), _Size, addressof(_Data[0])) {}
+
+    template <size_t _Size>
+    constexpr span(array<value_type, _Size>& _Data) noexcept
+        : _Mypair(_One_then_variadic_args_t(), _Size, _Data.data()) {}
+
+    template <size_t _Size>
+    constexpr span(const array<value_type, _Size>& _Data) noexcept
+        : _Mypair(_One_then_variadic_args_t(), _Size, _Data.data()) {}
+
+    template <class _OtherTy, size_t _OtherExtent>
+    constexpr span(const span<_OtherTy, _OtherExtent>& _Other,
+        enable_if_t<is_convertible_v<_OtherTy (*)[], element_type (*)[]>, nullptr_t> = nullptr) noexcept
+        : _Mypair(_One_then_variadic_args_t(), _OtherExtent, _Other.data()) {}
+
+    template <class _OtherTy>
+    constexpr span(const span<_OtherTy, dynamic_extent>& _Other,
+        enable_if_t<is_convertible_v<_OtherTy (*)[], element_type (*)[]>, nullptr_t> = nullptr) noexcept
+        : _Mypair(_One_then_variadic_args_t(), _Other.size(), _Other.data()) {}
+
+    constexpr span(const span&) noexcept = default;
+    constexpr span& operator=(const span&) noexcept = default;
+    // ~span() = default;
+
+    // [span.sub], span subviews
+    template <size_t _Count>
+    constexpr span<element_type, _Count> first() const noexcept {
+        if constexpr (_Extent != dynamic_extent) {
+            static_assert(_Count <= _Extent, "Count out of range in span::first()");
+        } else {
+#if _CONTAINER_DEBUG_LEVEL > 0
+            _STL_VERIFY(_Count <= size(), "Count out of range in span::first()");
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+        }
+        return {data(), _Count};
+    }
+
+    constexpr span<element_type, dynamic_extent> first(index_type _Count) const noexcept {
+#if _CONTAINER_DEBUG_LEVEL > 0
+        _STL_VERIFY(_Count <= size(), "Count out of range in span::first(Count)");

```

Comment:
The Standard doesn't capitalize arguments, so mentioning `span::first(Count)` seems strange. Occurs below.

Reply:
Done